src/probnum/randprocs/_random_process.py

```diff
@@ -48,11 +48,16 @@ class RandomProcess(Generic[_InputType, _OutputType], abc.ABC):
     def __init__(
         self,
         input_dim: IntArgType,
-        output_dim: IntArgType,
+        output_dim: Optional[IntArgType],
```

Comment:
```suggestion
        output_dim: Optional[IntArgType] = None,
```

Reply:
I omitted the default argument on purpose because I wanted to force `RandomProcess` subclass implementers to explicitly think about their output dimension. Would you change it anyway?